### PR TITLE
Fix C# client enum issue

### PR DESCRIPTION
**Pull request by wing328**
### PR checklist

- [ ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [ ] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.0.x`. Default: `master`.
- [ ] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

To fix https://github.com/OpenAPITools/openapi-generator/issues/635

I've only run `./bin/csharp-petstore.sh` for easier review (diff). I'll run other C# client scripts to update all samples later.

In addition to the fix, I've reformated the mustache templates to avoid trailing spaces and spaces in empty lines. 

cc @jimschubert @mandrean @albator1932 

Related PR: https://github.com/OpenAPITools/openapi-generator/pull/737

### CODE REVIEW

File: samples/client/petstore/csharp/OpenAPIClient/src/Org.OpenAPITools/Model/EnumTest.cs [hunk 1]
```diff
@@ -122,18 +120,16 @@ public enum EnumIntegerEnum
...
-            NUMBER_1_DOT_1 = 1,
-            
+            NUMBER_1_DOT_1 = 1.1,
+
             /// <summary>
             /// Enum NUMBER_MINUS_1_DOT_2 for value: -1.2
             /// </summary>
-            [EnumMember(Value = "-1.2")]
-            NUMBER_MINUS_1_DOT_2 = 2
+            NUMBER_MINUS_1_DOT_2 = -1.2
```
Comment (wing328) [inline samples/client/petstore/csharp/OpenAPIClient/src/Org.OpenAPITools/Model/EnumTest.cs]: @jimschubert @albator1932 For double/float value, do we need to restore the old behavior (using the string representation of the value)?

Comment (wing328) [reply samples/client/petstore/csharp/OpenAPIClient/src/Org.OpenAPITools/Model/EnumTest.cs]: Looks like it's not ok:
```
  Model\EnumTest.cs(126,30): error CS0266: Cannot implicitly convert type 'double' to 'int'. An explicit conversion exists (are you missing a cast?) [C:\projects\openapi-generator\samples\client\petstore\csharp\OpenAPIClient\src\Org.OpenAPITools\Org.OpenAPITools.csproj]
  Model\EnumTest.cs(131,36): error CS0266: Cannot implicitly convert type 'double' to 'int'. An explicit conversion exists (are you missing a cast?) [C:\projects\openapi-generator\samples\client\petstore\csharp\OpenAPIClient\src\Org.OpenAPITools\Org.OpenAPITools.csproj]
```
Let me fix it.

Comment (wing328) [reply samples/client/petstore/csharp/OpenAPIClient/src/Org.OpenAPITools/Model/EnumTest.cs]: Fixed via https://github.com/OpenAPITools/openapi-generator/pull/774/commits/ec5ca857b39bf4692b1d14107e0db731a84bea29

**Comment by jimschubert** [comment]
@wing328 I think this looks good. I can't tell if you're done making changes, though, so I haven't merged. Can you comment when it's ready to go?

**Comment by wing328** [comment]
@jimschubert yup, thanks for the review. Just merged it.

**Pull request merged**